lib/components/Selection control/Checkbox/index.tsx

```diff
@@ -0,0 +1,180 @@
+} from "@deriv/quill-icons";
+import { TMediumSizes } from "../../../types";
+import { Text } from "../../Typography";
+
+interface CheckboxProps
+    extends Omit<
+        ComponentProps<"input">,
+        "placeholder" | "defaultChecked" | "size"
+    > {
+    indeterminate?: boolean;
+    isLabelPaired?: boolean;
+    infoIconClassName?: string;
+    size?: TMediumSizes;
+    label: ReactNode | string;
+    labelClassName?: string;
+    onChange?: (
+        e:
+            | React.ChangeEvent<HTMLInputElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => void;
+    wrapperClassName?: string;
+}
+
+export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
+    (
+        {
+            checked = false,
+            disabled = false,
+            indeterminate = false,
+            isLabelPaired = false,
+            infoIconClassName,
+            size = "sm",
+            label,
+            labelClassName,
+            name,
+            onChange,
+            wrapperClassName,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [is_checked, setIsChecked] = useState(checked);
+        const [is_indeterminate, setIsIndeterminate] = useState(indeterminate);
+
+        useEffect(() => {
+            setIsChecked(checked);
+            setIsIndeterminate(indeterminate);
+        }, [checked, indeterminate]);
+
+        const onInputChange: React.ChangeEventHandler<HTMLInputElement> = (
+            e,
+        ) => {
+            e.stopPropagation();
+            if (is_indeterminate) {
+                setIsIndeterminate(!is_indeterminate);
+                setIsChecked(false);
+            } else {
+                setIsChecked(!is_checked);
+            }
+            onChange?.(e);
+        };
+
+        const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (e) => {
+            e.stopPropagation();
+            if (e.key === "Enter" || e.key === " ") {
+                if (is_indeterminate) {
+                    setIsIndeterminate(!is_indeterminate);
+                    setIsChecked(false);
+                } else {
+                    setIsChecked(!is_checked);
+                }
+                onChange?.(e);
+            }
+        };
+
+        const uncheckedIcon = is_indeterminate ? (
+            <StandaloneSquareMinusFillIcon
+                iconSize="sm"
+                className={clsx("quill-checkbox__icon", {
+                    "quill-checkbox__icon--disabled": disabled,
+                })}
+            />
+        ) : (
+            <StandaloneSquareBoldIcon
+                iconSize="sm"
+                className={clsx(
+                    "quill-checkbox__icon",
+                    "quill-checkbox__icon--default",
+                    {
+                        "quill-checkbox__icon--disabled": disabled,
+                    },
+                )}
+            />
+        );
+
+        const infoIcon =
+            size === "sm" ? (
```

Comment:
There is a difference between Standalone and Paired. 
`<StandaloneCircleInfoRegularIcon size='md'/> `will be 24x24,
`<LabelPairedCircleInfoMdRegularIcon /> `(also `md`) will be 24x16
Because of different width, I can't replace one icon type with another

Reply:
Same for sm